src/current/v25.2/vector-indexes.md

```diff
@@ -0,0 +1,323 @@
+
+Search accuracy is highly dependent on workload factors such as partition size, the number of `VECTOR` dimensions, how well the embeddings reflect semantic similarity, and how vectors are distributed in the dataset. Experiment with `vector_search_beam_size`, `min_partition_size`, and `max_partition_size` on a representative dataset to find the optimal setting for your workload.
+
+{{site.data.alerts.callout_success}}
+You can improve search accuracy for filtered queries by creating a [vector index with a prefix column](#create-and-query-a-vector-index).
+{{site.data.alerts.end}}
+
+#### Write performance
+
+Because a larger partition size leads to fewer partition splits and merges, it enables faster insert performance.
+
+To optimize writes, increase the partition size with `min_partition_size` and `max_partition_size`. A larger partition size leads to fewer partition splits and merges, resulting in faster insert performance.
+
+{{site.data.alerts.callout_info}}
+{% include {{ page.version.version }}/sql/vector-batch-inserts.md %}
+{{site.data.alerts.end}}
+
+## Example
+
+In the following example, a vector index with a prefix column is used to optimize searches on a `VECTOR` column in 512 dimensions. The example table is populated with 156,541 example rows, using a Python script to quickly insert the vectors.
+
+### Before you begin
+
+- Create a virtual `python3` environment and install `psycopg[binary]`:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    python3 -m venv ~/venv
+    source ~/venv/bin/activate
+    pip install psycopg[binary]
+    ~~~
+
+- Download the Python script and sample data:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    curl -O https://vector-examples.s3.us-east-2.amazonaws.com/fast_insert.py
+    curl -O https://vector-examples.s3.us-east-2.amazonaws.com/clip_embeddings_with_customers.csv
+    ~~~
+
+    Ensure that the `.py` and `.csv` files are located in the same directory.
+
+### Create and query a vector index
+
+1. Start a single-node cluster:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    cockroach start-single-node \
+    --insecure \
+    --listen-addr=localhost:26257 \
+    --http-addr=localhost:8080
+    ~~~
+
+1. In a separate terminal, [enable vector indexes](#enable-vector-indexes) on the cluster and session:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ sql
+    SET CLUSTER SETTING feature.vector_index.enabled = true;
+    ~~~
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ sql
+    SET sql_safe_updates = false;
```

Comment:
This step isn't necessary, since we're not creating the index on an already-populated table.